test/ibmq/test_ibmq_logger.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests related to logger setup via ``setup_logger()``."""
+
+import os
+import logging
+from unittest.mock import patch
+
+from qiskit.providers.ibmq.utils.utils import setup_logger
+
+from ..ibmqtestcase import IBMQTestCase
+
+
+class TestLogger(IBMQTestCase):
+    """Tests related to logger setup via ``setup_logger()``."""
+    log_level_env = 'QISKIT_IBMQ_PROVIDER_LOG_LEVEL'
+    log_file_env = 'QISKIT_IBMQ_PROVIDER_LOG_FILE'
+
+    @classmethod
+    def setUpClass(cls):
+        """Get a logger to test. Also, cache the initial log level and a file name.
+
+        Note:
+            The initial log level is cached in order to restore the log level
+            back to the original in `tearDown`.
+        """
+        cls.logger = logging.getLogger(__name__)
+        cls.initial_logger_level = cls.logger.level
+        cls.log_file = '{}.log'.format(cls.logger.name)
+
+    def tearDown(self):
+        """Restore the original log level and remove the logger handlers.
+
+        Note:
+            This essentially "resets" the logger, so it is set back into its
+            initial state again for each test.
+        """
+        self.logger.setLevel(self.initial_logger_level)
+        for handler in self.logger.handlers:
+            handler.close()
+        self.logger.handlers.clear()
+
+    def test_no_log_level(self):
+        """Test setting up a logger without a log level.
+
+        Note:
+            The log level should default to `NOTSET` when no level is specified.
+        """
+        default_level_not_set = logging.NOTSET
+
+        with patch.dict('os.environ'):
+            if self.log_level_env in os.environ:
+                del os.environ[self.log_level_env]
+            setup_logger(self.logger)
```

Comment:
That was my first thought, but I thought it would be too repetitive. I guess both solutions have their pros and cons. Although it's not an intensive task, it does seem worthwhile not having to do a `tearDown` after each test case if possible. Thanks!